lib/builder/builder.js

```diff
@@ -251,21 +273,22 @@ module.exports = {
 			}
 			return count;
 		}
-		const iProjectCount = projectCount(tree);
-		const buildLogger = log.createTaskLogger("🛠 ", iProjectCount);
+		const buildLogger = log.createTaskLogger("🛠 ", projectCount(tree));
 
 		function buildProject(project) {
 			let depPromise;
 			let projectTasks = selectedTasks;
+
+			// Build dependencies in sequence as it is far easier to detect issues and reduces
+			// side effects or other issues such as too many open files
 			if (buildDependencies) {
-				// Build dependencies in sequence as it is far easier to detect issues and reduces
-				// side effects or other issues such as too many open files
-				depPromise = project.dependencies.reduce(function(p, depProject) {
+				depPromise = (project.dependencies || []).filter(projectFilter).reduce(function(p, depProject) {
```

Comment:
`project.dependencies` must always be an array. A fallback should not be necessary.

See also https://github.com/SAP/ui5-project/pull/233